### PR TITLE
fix: overwriting of line #76

### PR DESCRIPTION
**Pull request by ahkrr**
This happened, because the `prompt` would be drawn at the `origin`, which was determined to be at the `position()?`, which was in the previous line. 
Now, move one line downwards before determining the `origin`.

### CODE REVIEW

File: src/engine.rs [hunk 1]
```diff
@@ -768,6 +768,8 @@ impl Reedline {
 
         let mut terminal_size = terminal::size()?;
 
+        self.stdout.queue(Print("\r\n"))?.flush()?;
```
Comment (jonathandturner) [inline src/engine.rs:771]: Do we always want to send a crlf? I was wondering if we may just want to see that we're at the bottom of the screen and scroll the screen up a line if so.

**Comment by ahkrr** [comment]
While thinking about this issue, I noticed that maybe `reedlines` use in `main.rs` should result in behavior like this
```
/home/hk/Dokumente/rus_test/reedline      06/24/2021 11:32:24 PM
〉ls
Our buffer: ls

/home/hk/Dokumente/rus_test/reedline      06/24/2021 11:32:26 PM
〉
```
because in `main.rs `
`line_editor.print_line(&format!("Our buffer: {}", buffer))?;`
is called, which prints a `\n` at the end placing the cursor in the next row. 
If an "application" would print a line without a trailing newline, this line would have been eaten by reedline previously.
The new changes now lead to the output looking like the example above.

**Comment by ahkrr** [comment]
Or should it be detected, if the position is already at (0, some_row) -> work from here, 
and if the position is ( column > 0, some_row) -> start in the next line

**Comment by ahkrr** [comment]
``` bash
oh-my-zsh
> printf "jo"
jo%
oh-my-zsh
> printf "jo\n"
jo
oh-my-zsh
>
```

**Comment by jonathandturner** [comment]
@ahkrr yeah, we could put the logic in to detect if we're not in the 0th column. Do you want to do that in a separate PR?

**Comment by ahkrr** [comment]
Lets do that in a separate PR.

**Comment by jonathandturner** [comment]
sounds good

**Pull request merged**